src/applications/caregivers/components/PreSubmitInfo/index.jsx

```diff
@@ -1,58 +1,60 @@
 import React, { useEffect, useState } from 'react';
+import { cloneDeep } from 'lodash';
 
 import SignatureCheckbox from './SignatureCheckbox';
 
 const PreSubmitCheckboxGroup = ({ onSectionComplete, formData, showError }) => {
-  const veteranLabel = `Enter Veteran\u2019s full name`;
-  const primaryLabel = `Enter Primary Family Caregiver applicant\u2019s full name`;
-  const secondaryOneLabel = `Enter Secondary Family Caregiver applicant\u2019s full name`;
-  const secondaryTwoLabel = `Enter Secondary Family Caregiver applicant\u2019s (2) full name`;
-
-  const [signatures, setSignature] = useState({
+  const veteranLabel = `Veteran\u2019s`;
+  const primaryLabel = `Primary Family Caregiver applicant\u2019s`;
+  const secondaryOneLabel = `Secondary Family Caregiver applicant\u2019s`;
+  const secondaryTwoLabel = `Secondary Family Caregiver (2) applicant\u2019s`;
+  const hasSecondaryOne = formData['view:hasSecondaryCaregiverOne'];
+  const hasSecondaryTwo = formData['view:hasSecondaryCaregiverTwo'];
+
+  const [signatures, setSignatures] = useState({
     [veteranLabel]: false,
     [primaryLabel]: false,
   });
 
-  const [secondaryCaregivers, setSecondaryCaregivers] = useState({
-    [secondaryOneLabel]: false,
-    [secondaryTwoLabel]: false,
-  });
   const unSignedLength = Object.values(signatures).filter(
     obj => Boolean(obj) === false,
   ).length;
 
+  // when there is no unsigned signatures set AGREED (onSectionComplete) to true
+  // if goes to another page (unmount), set AGREED (onSectionComplete) to false
   useEffect(
     () => {
-      if (!unSignedLength) {
-        onSectionComplete(true);
-      }
+      onSectionComplete(!unSignedLength);
 
-      if (unSignedLength) {
+      return () => {
         onSectionComplete(false);
-      }
+      };
+    },
 
-      const hasSecondaryOne =
-        formData?.secondaryOneFullName?.first &&
-        formData?.secondaryOneFullName?.last;
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
This ESLint line has been here, I just refactored this `useEffect`, I managed to remove  `react-hooks/exhaustive-deps` from all `useEffect`s besides this one because `onSectionComplete` function provided by `formBuilder` causes endless re-renders.